indico/util/user.py

```diff
@@ -181,3 +189,44 @@ def wrapper(*args, **kwargs):
         return fn(*args, **kwargs)
 
     return wrapper
+
+
+def create_user(user_form, handler, pending_user=None, is_admin=False):
+    """Creates a new user based on the data passed as argument"""
+    from indico.modules.users import user_management_settings
+    from indico.modules.users.models.users import User
+    from indico.modules.auth.util import send_notification_to_admins
+
+    user_data = user_form.data
+    minfo = HelperMaKaCInfo.getMaKaCInfoInstance()
+
+    if pending_user:
+        user = pending_user
+        user.is_pending = True
```

Comment:
wrong logic, you need `False` here (instead of creating a new user you take the pending one and convert it to a proper user by clearing the is_pending flag)